api/ims/ims-api.yaml

```diff
@@ -40,79 +38,138 @@ paths:
-     description: Function token refresh
-     responses:
+      operationId: refreshToken
+      description: Function that refreshes the token 
+      responses:
         '200':
-          description: Token refreshed successfully
+          description: Token successfully refreshed
           content:
-            'application/json':
-               schema:
+            application/json:
+              schema:
                 $ref: '#/components/schemas/Token'
         default:
           description: unexpected error
           content:
-            'application/json':
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /email/available:
+    get:
+      description: This endpoint compares user's entered email with existing emails and displays an error if there is the same in DB
+      operationId: emailAvailable
+      parameters:
+        - in: query
+          name: email
+          schema:
+            type: string
+            format: email
+          description: Compared email
+      responses:
+        '200':
+          description: Email is not occupied
+        '409':
+          description: User with this email has already registered
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /users/emailVerificationToken/verify:    
+    get:
+      description: To complete the registration you need to confirm your email with token
+      operationId: emailVerificationToken
+      parameters:
+        - in: query
+          name: token
+          schema:
+            type: string
+          description: Token that you need to complete the registration
+      responses:
+        '200':
+          description: Registration completed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SignupResponseWithToken'
+        default:
+          description: User did not confirm his email
+          content:
+            application/json:
               schema:
                 $ref: '#/components/schemas/Error'
 components:
   schemas:
     SignupRequest:
       type: object
-      required:
-        - firstName
-        - lastName
-        - email
-        - password
       properties:
         firstName:
           type: string
           maxLength: 255
-          description: Users first name
+          description: User's first name
         lastName:
           type: string
           maxLength: 255
-          description: Users last name
+          description: User's last name
         email:
           type: string
           format: email
-          description: Users email
+          description: User's email
         password:
           type: string
           format: password
-          description: Users password
-    LoginRequest:
-      type: object
+          description: User's password
       required:
+        - firstName
```

Comment:
why did you add these fields in the required section of this entity?
`
- firstName
- lastName
`

Reply:
I did not add them into the required section. They have already been there. And, as I know, you added them.

Reply:
And this is not my block of code.